cmd/minikube/cmd/docker-env.go

```diff
@@ -366,6 +373,66 @@ docker-cli install instructions: https://minikube.sigs.k8s.io/docs/tutorials/doc
 			if err != nil {
 				exit.Error(reason.IfSSHClient, "Error with ssh-add", err)
 			}
+			if co.Config.KubernetesConfig.ContainerRuntime == constants.Containerd {
+				//create a  temporary directory
+				tmpDirName := "/tmp/minikube-nerdctld"
+				if err := os.Mkdir(tmpDirName, os.ModePerm); err != nil && !os.IsExist(err) {
+					exit.Error(reason.IfSSHClient, "Error with creating "+tmpDirName, err)
+				}
+
+				// for containerd runtime using ssh, we need to install nerdctld and nerdctl within os
+				if err := download.DownloadNerdctl(tmpDirName); err != nil {
+					exit.Error(reason.IfSSHClient, "Error with downloading nerdctl", err)
+				}
+				if err := download.DownloadNerdctld(tmpDirName); err != nil {
+					exit.Error(reason.IfSSHClient, "Error with downloading nerdctld", err)
+				}
+
+				//copy these binaries to the path of the containerd node
+				co := mustload.Running(ClusterFlagValue())
+				runner := co.CP.Runner
+				src := newRemotePath(filepath.Join(tmpDirName, "nerdctl"))
+				dst := newRemotePath("/usr/local/bin/nerdctl")
+				if err := runner.Copy(copyableFile(&co, src, dst)); err != nil {
+					exit.Error(reason.InternalCommandRunner, "Fail to copy file nerctl", err)
+				}
+
+				src = newRemotePath(filepath.Join(tmpDirName, "nerdctld"))
+				dst = newRemotePath("/usr/local/bin/nerdctld")
+				if err := runner.Copy(copyableFile(&co, src, dst)); err != nil {
+					exit.Error(reason.InternalCommandRunner, "Fail to copy file nerctld", err)
+				}
+
+				//and set 777 to these files
+				if out, err := runner.RunCmd(exec.Command("sudo", "chmod", "777", "/usr/local/bin/nerdctl", "/usr/local/bin/nerdctld")); err != nil {
+					exit.Error(reason.InternalCommandRunner, fmt.Sprintf("Fail to giver nerdctl proper permission: %s", out.Output()), err)
+				}
+
+				//remove /var/run/docker.sock
+				if out, err := runner.RunCmd(exec.Command("sudo", "rm", "-rf", "/var/run/docker.sock")); err != nil {
```

Comment:
This should use `/var/run/nerdctl.sock`, instead of modifying the socket for a different container runtime

Reply:
Well I do wonder whether there exists a way to achieve this. When we are using docker client to connect to remore docker engine via ssh, we use `DOCKER_HOST="ssh://docker@127.0.0.1:49172" `, and I didn't find a way to tell docker client to use `/var/run/nerdctl.sock` instead of `/var/run/docker.sock`

Reply:
You need to configure it on the server, so that the environment variable is populated for a non-interactive shell.
This can be done in `~/.bashrc`, or in a wrapper for the `docker system dial-stdio` command that it is running.

The Docker documentation about it is somewhat vague, they only mention that it needs to be changed:
> To expose the Docker API socket through SSH, you need to make sure $DOCKER_HOST is set on the remote host.

https://docs.docker.com/engine/security/rootless/#expose-docker-api-socket-through-ssh

I think that for e.g. Podman Desktop, they just gave up on `ssh://` and do their own tunneling plus `unix://`.

There are also some legacy Docker clients (like Java), that don't support the new URLs introduced in 18.09.
While it is possible to do the tunneling over `minikube ssh` as well, I think it should be more of a last resort...

Basically it (the client) only supports the `tcp:` protocol for remote and the rest of it is just hacks upon hacks ?

Reply:
I see, I will have a try.

Reply:
Thanks! It works! I will make changes on this PR soon